SETUP/DEVELOPMENT.md

```diff
@@ -0,0 +1,79 @@
+# Code Development
+
+This doc is designed for users who want to develop on the DProofreaders (DP)
+codebase.
+
+## Introduction
+
+The DP code is almost entirely server-side PHP with a smattering of JS and
+CSS. It uses a MySQL backend database and requires a web server -- pgdp.net
+uses Apache but Nginx should work as well. See the [installation docs](INSTALL.md)
+for information on minimum middleware versions.
+
+The code does not use a PHP framework or an ORM. It was developed 20 years ago
+and has grown very organically over the years. There has been effort in the
+past decade to standardize things but you'll find many different patterns remain.
+
+## Development Environment
+
+The best current development environment is using the
+[DP development VM](https://www.pgdp.net/wiki/DP_Code_Development_VM) which
+comes with a recent release of the DP code, all required middleware, and some
+sample data. There is a desire to get a true local development environment --
+via vagrant or docker -- created but that has not yet been completed.
+
+### Tooling
+
+The code uses [Composer](https://getcomposer.org/) to manage PHP dependencies.
+Some of these dependencies are runtime dependencies, others are just used for
+development (phpstan, phpunit, etc). To install PHP dependencies (runtime
+and development) run the following in the repo base:
+```bash
+composer install
+```
+
+The code also uses [npm](https://www.npmjs.com/) for some _development only_
+dependencies, such as linting (eslint) and CSS creation (less). To install
+nodejs dependencies (runtime and development) run the following in the repo base:
+```bash
+npm install
+```
+
+## Development-related docs
+
+* [coding standards](CODE_STYLE.md)
+* [coding best practices](https://www.pgdp.net/wiki/DP_Code_Best_Practices)
+* [automated tests](tests/README.md)
+* [CSS / style documentation](../style/README.md)
+* [API documentation](../api/README.md)
+* [coding documentation](CODE_DOCS.md)
+* [graphs design](GRAPHS.md)
+* [Unicode support](UNICODE.md)
+
+## Organizing principals
+
+The code is loosely organized around the following ideas:
```

Comment:
This section is very useful to outline where things live in the repository :medal_sports: 

I wonder if it would be better to structure this a bit differently though as there seems to be two concepts being introduced concurrently in this section:
* The organizing principals, e.g. when to use `.php` vs `.inc`.
* An overview of the directories in the repo, e.g. `styles` contains all CSS files, `SETUP` the setup scripts + developer doc.

The latter may be slightly more readable as a list/tree of directories, followed by what they contain.

Reply:
Good suggestion. I've broken this apart, how is it now?

Reply:
That looks great, thanks!